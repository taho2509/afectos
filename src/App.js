import React, {Component} from 'react';
import './App.css';
import AfectosForm from './components/AfectosForm';
//i ll keep a bootstrap copy locally to avoid
// my connectivity issues but i think it would
// be better to put this in header
/*<link
    rel="stylesheet"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/latest/css/bootstrap.min.css"
    crossOrigin="anonymous"
/>*/

import './css/bootstrap.min.css';



class App extends Component {
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <p>
                        Una aplicacion sencilla para calcular el estado de animo de las personas en la manana, tarde y noche de cada dia
                    </p>
                </header>
                <AfectosForm/>
            </div>
        );
    }
}

export default App;
