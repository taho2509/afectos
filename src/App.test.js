import React from 'react';
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import App from "./App";
import {Form} from "react-bootstrap";


Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
    let shallowApp;
    const app = () => {
        if (!shallowApp) {
            shallowApp = shallow(
                <App/>
            );
        }
        return shallowApp;
    };

    beforeEach(() => {
        shallowApp = undefined;
    });

    // All tests will go here
    it('renders correctly', () => {
        const tree = app();
        expect(toJson(tree)).toMatchSnapshot();
    });

    it("always renders a div", () => {
        const divs = app().find("div");
        expect(divs.length).toBeGreaterThan(0);
    });

    it("always renders a header", () => {
        const header = app().find("header");
        expect(header.length).toBe(1);
    });

    it("always renders an `AfectosForm`", () => {
        const afectos = app().find("AfectosForm");
        expect(afectos.length).toBe(1);
    });

    describe("the rendered div", () => {
        it("contains everything else that gets rendered", () => {
            const divs = app().find("div");
            const wrappingDiv = divs.first();
            expect(wrappingDiv.children()).toEqual(app().children());
        });
    });

});
