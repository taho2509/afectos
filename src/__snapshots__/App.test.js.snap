// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <link
      crossOrigin="anonymous"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
      rel="stylesheet"
    />
    <p>
      Una aplicacion sencilla para calcular el estado de animo de las personas en la manana, tarde y noche de cada dia
    </p>
  </header>
  <AfectosForm />
</div>
`;
